src/push.rs

```diff
@@ -169,48 +174,81 @@ pub async fn build_profile_remotely(data: &PushProfileData<'_>, derivation_name:
 
 
     // copy the derivation to remote host so it can be built there
-    let copy_command_status = Command::new("nix").arg("copy")
-        .arg("-s")  // fetch dependencies from substitures, not localhost
-        .arg("--to").arg(&store_address)
-        .arg("--derivation").arg(derivation_name)
-        .env("NIX_SSHOPTS", ssh_opts_str.clone())
-        .stdout(Stdio::null())
-        .status()
-        .await
-        .map_err(PushProfileError::Copy)?;
+    let copy_command_status = {
+        let mut copy_command = Command::new("nix");
+        copy_command.arg("copy")
+            .arg("-s")  // fetch dependencies from substitures, not localhost
+            .arg("--to").arg(&store_address)
+            .arg("--derivation").arg(derivation_name)
+            .env("NIX_SSHOPTS", ssh_opts_str.clone());
+
+        debug!("copy command: {:?}", copy_command);
+
+        let mut child = copy_command.stderr(Stdio::piped()).stdout(Stdio::piped()).spawn().expect("failed to spawn nix copy command");
+
+        if let Some(pb) = &data.deploy_data.progressbar {
```

Comment:
Is there a chance to abstract away common parts that handle subprocess outputs?
Handling for `nix copy` and `nix build` looks completely identical and it might be easy to update only one of them when needed and forget about the second